File: backend/src/main/java/indiana/indi/indiana/dto/games/GameDetailsDto.java
package indiana.indi.indiana.dto.games;

import indiana.indi.indiana.dto.categories.CategoryDto;

import java.math.BigDecimal;
import java.util.List;

public record GameDetailsDto(
        Long id,
        String title,
        String details,
        String imageUrl,
        String gameFileUrl,
        Long authorId,
        List<CategoryDto> categories,
        BigDecimal price,
        Integer discountPercent,
        BigDecimal finalPrice,
        Integer rating,
        boolean isFavorite,
        boolean isInCart,
        boolean isPurchased
) {
}

File: frontend/src/types/games/GameDetailsDto.ts
import {Category} from "../categories/Category";

export type GameDetailsDto = {
    id: number;
    title: string;
    details: string;
    imageUrl: string;
    gameFileUrl: string;
    authorId: number;
    categories: Category[];
    price: number;
    discountPercent: number;
    finalPrice: number;
    rating: number;
    isFavorite: boolean;
    isInCart: boolean;
    isPurchased: boolean;
}
